app/src/main/java/org/puder/trs80/EditConfigurationFragment.java

```diff
@@ -87,13 +91,32 @@ public Preference findPreference(String name) {
             public boolean onPreferenceClick(Preference pref) {
                 String key = pref.getKey();
                 int requestCode;
+                Optional<String> dir;
+                Intent intent = new Intent(getActivity(), FileBrowserActivity.class);
+
                 if (key.equals("conf_cassette")) {
                     requestCode = 0;
+                    dir = configPersitence.getCasettePath();
                 } else {
                     requestCode = Integer.parseInt(key.substring(key.length() - 1));
+                    dir = configPersitence.getDiskPath(requestCode - 1);
+                }
+
+                if (dir.isPresent()) {
+                    dir = Optional.of(new File(dir.get()).getParent());
+                } else {
+                    try {
+                        ConfigurationManager configManager = ConfigurationManager.get(getActivity());
+                        dir = Optional.of(configManager.getEmulatorState(configId.get()).getBasePath());
+                    } catch (IOException e) {
```

Comment:
You should handle this exception somehow. E.g. showing an error or at least exiting the activity and logging something

Reply:
You're right. I see you handle a similar condition earlier in that file where something's wrong with the underlying configuration. For consistency, I now handle it in the same way: log the error and leave the Activity.

Reply:
The check for an existing file, and the ActionBar changes requested, have been implemented. 

I found a couple more hard-coded strings as well in the menu xml, so those have been moved out to strings and strings-de. 

Getting closer?